jaeger-spark-dependencies-elasticsearch/src/main/java/io/jaegertracing/spark/dependencies/elastic/ElasticsearchDependenciesJob.java

```diff
@@ -192,6 +192,7 @@ void run(String[] spanIndices, String[] depIndices,String peerServiceTag) {
             .map(new ElasticTupleToSpan())
             .groupBy(Span::getTraceId);
         List<Dependency> dependencyLinks = DependenciesSparkHelper.derive(traces,peerServiceTag);
+        // TODO the type names are deprecated (in ES7) and will be removed in a later release.
```

Comment:
This produces warning messages, however I think it's better than to require a conf parameter to avoid adding a type. Alternatively we could get the version from ES but we would have to use a separate client with security settings etc...

```
STORAGE=elasticsearch ES_NODES=http://localhost:9200 java -jar jaeger-spark-dependencies/target/jaeger-spark-dependencies-0.0.1-SNAPSHOT.jar                                     5:23 
19/08/13 17:23:06 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
19/08/13 17:23:06 INFO ElasticsearchDependenciesJob: Running Dependencies job for 2019-08-13T00:00Z, reading from jaeger-span-2019-08-13 index, result storing to jaeger-dependencies-2019-08-13
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:08 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:09 WARN Resource: Detected type name in resource [jaeger-dependencies-2019-08-13/dependencies]. Type names are deprecated and will be removed in a later release.
19/08/13 17:23:09 INFO ElasticsearchDependenciesJob: Done, 8 dependency objects created
```

Reply:
I have added a mechanism to get the ES version and appropriately add type for older versions. However, the warn message is still present when used with older ES versions.